src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,146 @@
+
+            statement.setString(FIRST_QUESTION_MARK_INDEX,
+                    book.getTitle());
+            statement.setBigDecimal(SECOND_QUESTION_MARK_INDEX,
+                    book.getPrice());
+            int end = statement.executeUpdate();
+            if (end < 1) {
+                throw new DataProcessingException("Failed to save book");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_QUESTION_MARK_INDEX, Long.class);
+                book.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(TAKE_BOOK_FROM_DB)) {
+
+            statement.setLong(FIRST_QUESTION_MARK_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+
+                Long idFromBook = resultSet.getObject(ID,
+                        Long.class);
+                String titleFromBook = resultSet.getString(TITLE);
+                BigDecimal priceFromBook = resultSet.getObject(PRICE,
+                        BigDecimal.class);
+
+                Book book = new Book(idFromBook, titleFromBook, priceFromBook);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(TAKE_BOOKS_FROM_DB)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+
+                Long idFromBook = resultSet.getObject(ID,
+                        Long.class);
+                String titleFromBook = resultSet.getString(TITLE);
+                BigDecimal priceFromBook = resultSet.getObject(PRICE,
+                        BigDecimal.class);
+
+                Book book = new Book(idFromBook, titleFromBook, priceFromBook);
+                books.add(book);
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_BOOK_FROM_DB)) {
+
+            statement.setLong(FIRST_QUESTION_MARK_INDEX, id);
+            statement.executeUpdate();
+            return true;
```

Comment:
It should be done like this (or even inlined into 1 line only):
```suggestion
            int updatedRows = statement.executeUpdate();
            return updatedRows > 0;
```